src/transport/BUILD.gn

```diff
@@ -14,8 +14,23 @@
 
 import("//build_overrides/chip.gni")
 import("//build_overrides/nlio.gni")
+import("//build_overrides/pigweed.gni")
+import("${chip_root}/build/chip/buildconfig_header.gni")
 import("${chip_root}/src/ble/ble.gni")
 
+declare_args() {
+  # When enabled traces messages using pw_trace.
+  chip_enable_transport_trace = false
+}
+
+buildconfig_header("transport_buildconfig") {
+  header = "TransportBuildConfig.h"
+  header_dir = "transport"
+
+  defines =
+      [ "CHIP_CONFIG_TRANSPORT_TRACE_ENABLED=${chip_enable_transport_trace}" ]
```

Comment:
Do we really want a separate knob for every subsystem?

It's possibly useful, but adds complexity. I would suggest starting simpler with a global knob.

Reply:
Done